src/Compilers/CSharp/Portable/Symbols/ConstructedNamedTypeSymbol.cs

```diff
@@ -54,6 +60,11 @@ internal ConstructedNamedTypeSymbol(NamedTypeSymbol constructedFrom, ImmutableAr
             Debug.Assert(constructedFrom.Arity != 0);
         }
 
+        protected override NamedTypeSymbol WithTupleDataCore(TupleUncommonData newData)
+        {
+            return new ConstructedNamedTypeSymbol(_constructedFrom, _typeArgumentsWithAnnotations, IsUnboundGenericType) { _lazyTupleData = newData };
```

Comment:
>return new ConstructedNamedTypeSymbol(_constructedFrom, _typeArgumentsWithAnnotations, IsUnboundGenericType) { _lazyTupleData = newData }; [](start = 12, length = 138)

It looks like all the `WithTupleDataCore()` implementations simply clone the instance and set `_lazyTupleData`.

Would it make sense to rename this method `Clone()` with no arguments and have the base class caller set`_lazyTupleData` instead? As a side benefit, `_lazyTupleData` could be marked as `private`.

Reply:
Or alternatively, add an optional `TupleUncommonData` argument to the various constructors and change `_lazyTupleData` to a `private readonly` field.

---
In reply to: [355212799](https://github.com/dotnet/roslyn/pull/39370#discussion_r355212799) [](ancestors = 355212799)